packages/batch-poster-monitor/index.ts

```diff
@@ -325,13 +349,24 @@ const monitorBatchPoster = async (childChainInformation: ChainInfo) => {
   const latestChildChainBlockNumber = await childChainClient.getBlockNumber()
 
   if (!sequencerInboxLogs || sequencerInboxLogs.length === 0) {
-    // No SequencerInboxLog in the last 12 hours (time hardcoded in getDefaultBlockRange)
-    // We compare the "latest" and "safe" blocks
+    // No SequencerInboxLog in the last 24 hours (time hardcoded in getDefaultBlockRange)
+    // We compare the "latest" and "safe" blocks, and check if any of the pending blocks contain any user-transactions
     // NOTE: another way of verifying this might be to check the timestamp of the last block in the childChain chain to verify more or less if it should have been posted
     const latestChildChainSafeBlock = await childChainClient.getBlock({
       blockTag: 'safe',
     })
-    if (latestChildChainSafeBlock.number < latestChildChainBlockNumber) {
+
+    const blocksPendingToBePosted =
+      latestChildChainSafeBlock.number < latestChildChainBlockNumber
+
+    const doPendingBlocksContainUserTransactions =
+      await checkForUserTransactionBlocks({
+        fromBlock: Number(latestChildChainSafeBlock.number.toString()) + 1, // start checking AFTER the latest 'safe' block
```

Comment:
Actually `latestChildChainSafeBlock.number` is a `bigint` type - we convert it to number so that we can iterate over it inside the function `checkForUserTransactionBlocks `

Reply:
I think we could skip the `toString` part:
```suggestion
        fromBlock: Number(latestChildChainSafeBlock.number + 1n), // start checking AFTER the latest 'safe' block
```